### DISCUSSION TITLE
Auto-login user to data source

### DISCUSSION THREAD

**Discussion by slice-viditp** [cloudera/hue#3320] [Questions] [answered]
Hey !
Is there some way to autofill the username while connecting to a data source ?

<img width="615" alt="image" src="https://github.com/cloudera/hue/assets/113665661/df4b34c5-b0f2-49ad-9570-bfecdc19843b">

I basically want to turn off this user prompt and have the user's login name automatically fill in the username field without giving an option to change.

Trying to set this configuration for Trino connector.

One configuration I tried was to set the interpreter as following 

`  {"url": "jdbc:trino://trino.url:443/glue", "driver": "io.trino.jdbc.TrinoDriver" , "user" : $USER} 
` 

But this crashed the UI on login and resulted in a blank white screen with error 403.

It would be great if someone can help here !

**Comment by Harshg999** [comment, marked as answer]
Hey @slice-viditp, this looks like a very specific use-case and as of now we don't support anything like this from config/backend or UI side.

Maybe there are some UI tricks available to auto-fill the username for this data source everytime this pop-up comes up (by scraping the tags or using some extensions).
Or maybe you can try playing around the `sql_alchemy.py` + UI part to extract and send this username (not via any config but from `User` object) for Trino and plug it in the data source field for your use-case.

These are some of the ideas on top of my mind. Hope this answer helps!

**Comment by slice-viditp** [comment]
@Harshg999 Got it , Thanks for the reply.
